adapters/openweb/openweb.go

```diff
@@ -0,0 +1,179 @@
+package openweb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type openwebImpExt struct {
+	OpenWeb openrtb_ext.ExtImpOpenWeb `json:"openweb"`
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	totalImps := len(request.Imp)
+	errors := make([]error, 0, totalImps)
+	sourceIdToImpIds := make(map[int][]int)
+
+	for i := 0; i < totalImps; i++ {
+
+		sourceId, err := validateImpression(&request.Imp[i])
+
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if _, ok := sourceIdToImpIds[sourceId]; !ok {
+			sourceIdToImpIds[sourceId] = make([]int, 0, totalImps-i)
+		}
+
+		sourceIdToImpIds[sourceId] = append(sourceIdToImpIds[sourceId], i)
+
+	}
+
+	totalReqs := len(sourceIdToImpIds)
+	if 0 == totalReqs {
+		return nil, errors
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	reqs := make([]*adapters.RequestData, 0, totalReqs)
+
+	imps := request.Imp
+	request.Imp = make([]openrtb2.Imp, 0, len(imps))
+	for sourceId, impIds := range sourceIdToImpIds {
```

Comment:
Please add a JSON test for this logic where a request has multiple imps with multiple source IDs and they are properly grouped together as requests for your bidder service